src/components/external/DisputeManagement/components/DefendDisputeFlow/DefendDisputeFileUpload.tsx

```diff
@@ -90,18 +185,54 @@ export const DefendDisputeFileUpload = () => {
                     </ul>
                 </Card>
                 <div className={'adyen-pe-defend-dispute-file-uploader__container'}>
-                    {applicableDocuments?.map((document: IDisputeDefenseDocument) => (
-                        <FileInput
-                            ref={ref}
-                            key={document}
-                            required={document.requirementLevel === 'REQUIRED'}
-                            onChange={file => {
-                                if (file[0]) addFileToDefendPayload(document.documentTypeCode, file[0]);
-                            }}
-                        >
-                            {i18n.get(document.documentTypeCode as TranslationKey)}
-                        </FileInput>
-                    ))}
+                    <div className="adyen-pe-defend-dispute-document-upload-box">
+                        <div className="adyen-pe-defend-dispute-document-upload-box__required-documents">
+                            {requiredDocuments?.map(document => {
+                                return (
+                                    <DefendDocumentUpload
+                                        key={document}
+                                        document={document}
+                                        ref={ref}
+                                        addFileToDefendPayload={addFileToDefendPayload}
+                                        isRequired
+                                    />
+                                );
+                            })}
+                        </div>
```

Comment:
Only render the group of required documents if there is at least one.

```suggestion
                        {requiredDocuments.length ? (
                            <div className="adyen-pe-defend-dispute-document-upload-box__required-documents">
                                {requiredDocuments.map(document => {
                                    return (
                                        <DefendDocumentUpload
                                            key={document}
                                            document={document}
                                            ref={ref}
                                            addFileToDefendPayload={addFileToDefendPayload}
                                            isRequired
                                        />
                                    );
                                })}
                            </div>
                            ) : null}
```

Reply:
✅

Reply:
This doesn't look like it has been done — I think you should double-check.